server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -1085,6 +1085,11 @@ protected void parseCreateField(ParseContext context) throws IOException {
         }
     }
 
+    @Override
+    protected Object parseSourceValue(Object value) {
```

Comment:
Would it be useful to explicitly return `Number` from the method?

Reply:
👍 in the follow-up PR I can make sure these return the most specific type possible.

Reply:
:+1: